src/controllers/dapps/dapps.ts

```diff
@@ -112,6 +120,54 @@ export class DappsController extends EventEmitter implements IDappsController {
       }
     })
 
+    this.#requests.onUpdate(async () => {
+      try {
+        if (
+          this.#requests.actions.currentAction &&
+          this.#requests.actions.currentAction.type === 'dappRequest' &&
+          this.#requests.actions.currentAction.userRequest.action.kind === 'dappConnect'
+        ) {
+          const { session } = this.#requests.actions.currentAction.userRequest
+          const dapp = await this.#buildDapp({
+            id: session.id,
+            name: session.name,
+            url: session.origin,
+            icon: session.icon,
+            chainId: 1,
+            isConnected: false
+          })
+          if (!this.dappToConnect || this.dappToConnect.id !== dapp.id) {
+            this.dappToConnect = dapp
+            this.#phishing.updateDappsBlacklistedStatus([dapp.url], (blacklistedStatus) => {
+              if (this.dappToConnect) {
+                this.dappToConnect.blacklisted = blacklistedStatus[dapp.id]
+                this.emitUpdate()
+              }
+
+              const existingDapp = this.#dapps.get(dapp.id)
+              if (existingDapp && existingDapp.blacklisted !== blacklistedStatus[dapp.id]) {
+                this.#dapps.set(dapp.id, {
+                  ...existingDapp,
+                  blacklisted: blacklistedStatus[dapp.id]
+                })
+                this.emitUpdate()
+              }
+            })
+            this.emitUpdate()
+          }
+
+          return
+        }
+
+        if (this.dappToConnect) {
+          this.dappToConnect = null
+          this.emitUpdate()
+        }
+      } catch (err) {
+        console.error('Error in DappsController while updating the dappToConnect:', err)
+      }
+    })
```

Comment:
I don't really like it how the dapps controller uses requests and requests uses the dapp controller :thinking: Wouldn't it be cooler to pass a callback function (like getDapp) to the requests controller - `onSetCurrentAction`? The function will run this exact code. That way:
- There won't be a two way dependency
- This code won't run for every emit update (tho this one is very minor)